### PR TITLE
Make simplified download button (Fixes #8860)

### LINKED ISSUE #8860: [Epic] Make a simplified version of the download button
## Description

The current download button helper is versatile in the way that it supports multiple products (release, beta, nightly etc) and can also have the option to download directly (vs link to `/download/thanks/`). A large proporation of use-cases on the website exist to just serve the main Firefox release verson however, and all link directly to the `/download/thanks/` page.

For those use-cases, the HTML could probably be simplified to just the following:

```html
<div class="mzp-c-button-download-container">
  <a href="/%LOCALE%/firefox/download/thanks/" class="mzp-c-button mzp-t-product" data-direct-link="https://download.mozilla.org/?product=firefox-stub&amp;os=win&amp;lang=%LOCALE%">
    Download Firefox
  </a>
  <small class="mzp-c-button-download-privacy-link">
    <a href="/%LOCALE%/privacy/firefox/">Firefox Privacy Notice</a>
  </small>
</div>
```

Note that the `data-direct-link` attribute would still need to exist, but only to serve Windows builds to legacy IE users (IE9 and below).

The other thing that's omitted here is all the conditional markup for [unsupported messaging]( https://github.com/mozilla/bedrock/blob/master/bedrock/firefox/templates/firefox/includes/download-button.html#L42-L59), such as:

```html
<p class="unsupported-download">
  Your system doesn't meet the <a href="/firefox/system-requirements/">requirements</a> to run Firefox.
</p>
 ```

Having all this content in every download button, often 2 or 3 times per page, adds a lot of hidden content that's often not needed to the user. This can add to page weight, make our content harder to understand for crawlers, and can also surface as confusing messages in search results (see https://github.com/mozilla/bedrock/issues/8739).

Instead of having this content in every page of the website, we could move it to `/download/thanks/`. This page could then serve as the main place for such conditional messaging, allowing us to use a simplified download button helper for most of the site.

### PR DESCRIPTION
**Pull request by alexgibson**
## Description
- Adds a new simplified download button helper.
- Implements the simple button on the following pages / areas:
  - Download button in the global navigation.
  - Download buttons on http://localhost:8000/en-US/firefox/new/
  - Download buttons on home pages:
    - http://localhost:8000/en-US/
    - http://localhost:8000/de/
    - http://localhost:8000/fr/
    - http://localhost:8000/es-ES/

## Issue / Bugzilla link
#8860

## Testing
Demo: https://www-demo2.allizom.org/en-US/firefox/new/

- [x] Clicking download on a desktop browser navigates to /thanks and initiates a download as expected.
- [x] Clicking download in an old IE browser initiates a download via a popup, before navigating to /thanks as expected.
- [x] Clicking download on Android / iOS navigated to /thanks before redirecting to the app/play store.

Successful test run: https://gitlab.com/mozmeao/www-config/-/pipelines/188897807

### CODE REVIEW

**Comment by alexgibson** [comment]
This needs to be reviewed and merged after https://github.com/mozilla/bedrock/pull/9274

**Comment by alexgibson** [comment]
This is ready for code review, but do-not-merge pending GA testing / verification.

File: bedrock/firefox/templatetags/helpers.py
Comment (stephaniehobson) [inline bedrock/firefox/templatetags/helpers.py:194]: How does this work? If a `funnelcake_id` is defined somewhere on the page which calls the button this picks that up?

File: bedrock/firefox/templatetags/helpers.py
Comment (stephaniehobson) [inline bedrock/firefox/templatetags/helpers.py:210]: The original `download_firefox` has some documentation about what these mean. If this function will eventually replace that one it would be nice to preserve the documentation.

**Review by stephaniehobson — COMMENTED** [review]
I haven't started looking at the JS yet but here are a few comments from my initial pass.

File: media/js/base/mozilla-utils.js
Comment (stephaniehobson) [inline media/js/base/mozilla-utils.js:76]: There isn't a 1:1 mapping for the version defined by the old button to the version defined in the JS here. (For example win64, win-msi, and standard will be reported as something else) Is that okay?

**Comment by stephaniehobson** [comment]
I can't wait to see the impact implementing this in the nav bar has on performance! 🔥

File: bedrock/firefox/templatetags/helpers.py
Comment (alexgibson) [reply bedrock/firefox/templatetags/helpers.py:194]: If there's a funnelcake param in the url (e.g. ?f=123) then that will get passed to the /thanks page (e.g. /download/thanks/?f=123). I'll add a comment to document what it does.

File: bedrock/firefox/templatetags/helpers.py
Comment (alexgibson) [reply bedrock/firefox/templatetags/helpers.py:210]: I don't think this will replace the full download button entirely as it's needed to download pre-release products still, but good call to add some more docs 👍

File: media/js/base/mozilla-utils.js
Comment (alexgibson) [reply media/js/base/mozilla-utils.js:76]: Win64 downloads have never been enabled via the dl button (since the stub installer handles detection), and I don't think we'd ever serve msi installers via /thanks. So I think this is OK.

File: bedrock/firefox/templates/firefox/includes/download-button-thanks.html
Comment (alexgibson) [reply bedrock/firefox/templates/firefox/includes/download-button-thanks.html:1]: You can already add a custom button class and ID to the link itself. I don't think it makes sense to have another set for the wrapper, so if it's more useful I'll move them both to the top HTML element?

File: bedrock/firefox/templates/firefox/includes/download-button-thanks.html
Comment (alexgibson) [reply bedrock/firefox/templates/firefox/includes/download-button-thanks.html:4]: It's not strictly needed, but I think having a singular identifiable class for the link (rather than needing `.c-button-download-thanks > a` makes sense. For compatibility purposes `download-link` wfm 👍

File: bedrock/firefox/templates/firefox/includes/download-button-thanks.html
Comment (alexgibson) [reply bedrock/firefox/templates/firefox/includes/download-button-thanks.html:1]: Ok, we need to have a custom button class on the link so we can do things like `mzp-t-secondary`. It might be potentially confusing to have a different param and class for the wrapper, so what I propose is to make the ID belong to the top level element. Does that work?

File: bedrock/firefox/templates/firefox/includes/download-button-thanks.html
Comment (alexgibson) [reply bedrock/firefox/templates/firefox/includes/download-button-thanks.html:14]: It looks like the new button was used on 2/5 of those .scss files. I've updated the ones that were effected. nice catch 👍

**Comment by alexgibson** [comment]
Thanks @stephaniehobson - I pushed a commit that addresses some feedback. I also added some docs for the download button to try and help people understand which one to use and when.

**Comment by alexgibson** [comment]
@pmac r on the python bits here? I'm mostly wondering if I've overlooked anything that should still be included in the newer helper.

File: bedrock/firefox/templatetags/helpers.py
Comment (pmac) [inline bedrock/firefox/templatetags/helpers.py:197]: This is not something we should fix in this PR, but the `version` parameter to the `get_download_url` isn't even used in the function anymore. It's a vestige of the old way of generating direct download links before we only used the `*-latest` bouncer aliases. This doesn't hurt anything, but you could also just pass in `None` for `version` and it would work the same.

File: bedrock/firefox/tests/test_helpers.py
Comment (pmac) [inline bedrock/firefox/tests/test_helpers.py:360]: style nit: superfluous parens

File: bedrock/firefox/tests/test_helpers.py
Comment (pmac) [inline bedrock/firefox/tests/test_helpers.py:335]: style nit: superfluous parens

**Review by pmac — APPROVED** [review]
Looks good! Just a couple of small notes, but 👍 from me.

File: bedrock/firefox/templates/firefox/includes/download-button-thanks.html
Comment (stephaniehobson) [reply bedrock/firefox/templates/firefox/includes/download-button-thanks.html]: ID on top-level +1.

I think that's best until we can have parent selectors in CSS.

File: bedrock/firefox/templatetags/helpers.py
Comment (stephaniehobson) [reply bedrock/firefox/templatetags/helpers.py:194]: So we'd still have to use the other button for a campaign page that has a funnel cake? (That's okay, I just want to understand).

Comment (alexgibson) [reply bedrock/firefox/templatetags/helpers.py:194]: No, funnel cakes should work for both button helpers 🙂

File: media/js/base/mozilla-utils.js
Comment (stephaniehobson) [reply media/js/base/mozilla-utils.js:76]: There's not a lot of them but win64, win64-msi are currently being [reported in GA](https://analytics.google.com/analytics/web/#/report/content-event-events/a36116321w64087799p65789850/explorer-table.secSegmentId=analytics.customDimension27&explorer-table.plotKeys=%5B%5D&explorer-table.rowCount=250&explorer-table.filter=win). I don't see standard though, even though it is documented in the read the docs.

Comment (alexgibson) [reply media/js/base/mozilla-utils.js:76]: There’s a modal on /windows that enables downloading specific builds. On /browsers/windows-64-bit/ my guess is they are following the link to the /all page.

I don’t think any of these events are coming from download buttons since those links have never been enabled to appear.

Comment (stephaniehobson) [reply media/js/base/mozilla-utils.js:76]: If you[ filter to /firefox/new](https://analytics.google.com/analytics/web/#/report/content-event-events/a36116321w64087799p65789850/explorer-table.plotKeys=%5B%5D&explorer-table.secSegmentId=analytics.customDimension27&_r.drilldown=analytics.eventCategory:~2Ffirefox~2Fnew~2F%20Interactions,analytics.eventAction:Firefox%20Download/) you can see they are being triggered on /new 🤷‍♀️

Comment (stephaniehobson) [reply media/js/base/mozilla-utils.js:76]: Is that the modal there too?

Comment (stephaniehobson) [reply media/js/base/mozilla-utils.js:76]: Hm, also seeing them on [/firefox](https://analytics.google.com/analytics/web/#/report/content-event-events/a36116321w64087799p65789850/_r.drilldown=analytics.eventCategory:~2Ffirefox~2F%20Interactions&explorer-table.secSegmentId=analytics.customDimension27&explorer-table.plotKeys=%5B%5D/). No modal there.

Comment (alexgibson) [reply media/js/base/mozilla-utils.js:76]: The older version of /new has the modal yes. Are these session based events? That may explain things. Still think it this is likely a red herring.

File: bedrock/firefox/templates/firefox/includes/download-button-thanks.html
Comment (stephaniehobson) [inline bedrock/firefox/templates/firefox/includes/download-button-thanks.html:5]: There's a change in the GA reporting. This is showing up with the Event Label "Download for unavailable". I think it is because `data-download-os` is defined on the old button but not this one.

Comment (stephaniehobson) [reply bedrock/firefox/templates/firefox/includes/download-button-thanks.html:5]: Otherwise GA reporting looks to be the same in Chrome/MacOS:

| Prod     | Demo |
| --- | --- |
| Screen Shot 2020-09-15 at 07 22 12 | Screen Shot 2020-09-15 at 07 24 19 |

Comment (alexgibson) [reply bedrock/firefox/templates/firefox/includes/download-button-thanks.html:5]: Strange, because I'm adding `data-download-os` via JS (since the value is dynamic)

Comment (alexgibson) [reply bedrock/firefox/templates/firefox/includes/download-button-thanks.html:5]: I can see `data-download-os` on the link when viewing source on demo:

image

Comment (alexgibson) [reply bedrock/firefox/templates/firefox/includes/download-button-thanks.html:5]: Oh wait - it's `data-display-os` (face palm). At least that's a sinple fix :)

File: docs/download-buttons.rst
Comment (stephaniehobson) [inline docs/download-buttons.rst:8]: 📄 💯

File: bedrock/firefox/templates/firefox/includes/download-button-thanks.html
Comment (stephaniehobson) [reply bedrock/firefox/templates/firefox/includes/download-button-thanks.html:5]: Oh, I was viewing source. Then I don't know what's causing the difference. Are you able to look into it?

Comment (alexgibson) [reply bedrock/firefox/templates/firefox/includes/download-button-thanks.html:5]: Yeah I think I've got it - typo (see comment above ^) :)

Comment (stephaniehobson) [reply bedrock/firefox/templates/firefox/includes/download-button-thanks.html:5]: LOL, I made the same mistake in my comment 🤦‍♀️

Comment (alexgibson) [reply bedrock/firefox/templates/firefox/includes/download-button-thanks.html:5]: Updated data attribute to the correct name

**Comment by alexgibson** [comment]
Leaving do-not-merge in place until adswerve have taken a look in GA

**Comment by alexgibson** [comment]
Ok, adswerve have done some QA and confirmed things look good in GA. I'm going to remove the DNM label from the PR, but we should probably wait until Monday to merge and push to prod (pending final review approval).

**Review by stephaniehobson — APPROVED** [review]
R+ 🦋

**Pull request merged**
